library/alloc/src/collections/btree/map/tests.rs

```diff
@@ -1837,6 +1837,23 @@ fn test_vacant_entry_key() {
     a.check();
 }
 
+#[test]
+fn test_vacant_entry_no_insert() {
+    let mut a = BTreeMap::<&str, ()>::new();
+    let key = "hello there";
+
+    assert!(a.is_empty());
+    match a.entry(key) {
+        Occupied(_) => panic!(),
+        Vacant(e) => {
+            assert_eq!(key, *e.key());
+            drop(e);
```

Comment:
Here `drop` means: do nothing on the entry `e`, just drop it. I'd like to emphasize there is no action to do on the entry, no special reason.

Reply:
But how is it different from all the other places where a variable goes out of scope, or even gets an underscored name? When we haven't done everything we might possibly do with a variable? I assume you'd prefer the same in `test_occupied_entry_key`?

One thing I see now is that `insert` consumes the variable, and not very many types have such members. `drop` is an alternative way to consume it. In fact, do we even have a test case checking what happens to the map if a vacant entry is dropped without insertion? Yes, in `test_entry`, I think. Being explicit about it might help to see that. But it still feels a bit random to try that, if the compiler cannot possibly tell you when you miss it.

Reply:
> I assume you'd prefer the same in `test_occupied_entry_key`?

I didn't noticed that. Keeping consistent is better.